File: codes/sates_test_cs/testcode.cs
//------------------------------------------------------------------------------
// Copyright (C) 2018, Seokhwan Kim (kim at seokhwan.net)
// This file is part of "the SATES"
// For conditions of distribution and use, see copyright notice in 
// sates.core.doc.cs
//------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Text;


namespace sates.test.cs
{
    /** \addtogroup CODE
    *  @{
    */
    /** \addtogroup sates
    *  @{
    */
    /** \addtogroup test
    *  @{
    */
    /** \addtogroup cs
    *  @{
    */
    public enum TEST_RESULT
    {
        INVALID = 1,
        OK = 20180417,
        NG = 0,
    }

    public abstract class testcode
    {
        public TEST_RESULT result { get; set; }
        public Queue<string> err_log { get; set; }
        public string test_case_name { get; set; }

        protected testcode()
        {
            test_case_name = this.GetType().Name;
            err_log = new Queue<string>();
            testcode_list.add_testcode(this);
        }
        public abstract void init();
        public abstract void run();
        public abstract void terminate();

    }
    /** @} */
    /** @} */
    /** @} */
    /** @} */
}

File: codes/sates_test_java/src/com/sates/test/java/testcode.java
package com.sates.test.java;

public abstract class testcode
{
    public TEST_RESULT result;
    public java.util.Queue<String> err_log;
    public String test_case_name;

    protected  testcode()
    {
        test_case_name = this.getClass().getName();
        int lastdot = test_case_name.lastIndexOf(".");
        int lastindex = test_case_name.length();
        test_case_name = test_case_name.substring(lastdot + 1,  lastindex);
        err_log = new java.util.ArrayDeque<>();
    }

    public abstract void init();
    public abstract void run();
    public abstract void terminate();
}
